CRM/Core/BAO/ActionSchedule.php

```diff
@@ -546,6 +546,12 @@ protected static function sendReminderSms($tokenRow, $schedule, $toContactID) {
       return ["sms_phone_missing" => "Couldn't find recipient's phone number."];
     }
 
+    // dev/core#369 If an SMS provider is deleted then the relevant row in the action_schedule_table is set to NULL
+    // So we need to exclude them.
+    if (CRM_Utils_System::isNull($schedule->sms_provider_id)) {
+      return ["sms_provider_missing" => "SMS Provider is NULL in database cannot send reminder"];
```

Comment:
Is this message sent to a log or the end user?  It should read: "SMS reminder cannot be sent because the SMS provider has been deleted."

Reply:
@seamuslee001 my instinct is we should thow an exception here & the calling function should catch & handle it

Reply:
@andrew-cormick-dockery its going into the civicrm_aciton_log wherever the action schedule logs are  @eileenmcnaughton possibly but that isn't consistent with how this is written. note that in L546 we pass back an array as well if we can't find a legit phone number

Reply:
@seamuslee001 well I'm not thrilled with that idea - but that's not a complete abort as the missing provider is is it?

Reply:
@eileenmcnaughton well it aborts for that reminder, like in this case here we abort because we cannot find the relevant SMS provider, In that case it aborts because it cannot find a relevant phone number to send it to.

Reply:
OK - I'm not liking it but I guess I can ignore it

Reply:
The calling function here https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/BAO/ActionSchedule.php#L307 is expecting an array yo be returned for *reasons* and the only things in the array are any errors that have occurred.